sklearn/preprocessing/_discretization.py

```diff
@@ -198,7 +198,7 @@ def __init__(
         self.random_state = random_state
 
     @_fit_context(prefer_skip_nested_validation=True)
-    def fit(self, X, y=None, sample_weight=None):
+    def fit(self, X, y=None, sample_weight=None, use_weights_in_resampling=True):
```

Comment:
Based on the results of the analysis conducted in the notebooks linked in the PR discussion, we can now remove `use_weights_in_resampling` and always use the weights for resampling.